docs/coding-guidelines.md

```diff
@@ -0,0 +1,69 @@
+# Coding Guidelines
+
+This living document serves to prescribe coding guidelines specific to the WooCommerce Blocks project. Base coding guidelines follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+The following sections outline additional patterns and conventions used in the Blocks project.
+
+## CSS
+
+### CSS Class Names
+
+To avoid class name collisions, class names must adhere to the following guidelines, based on the [BEM methodology](https://en.bem.info/methodology/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+#### Prefixing
+
+As a WordPress plugin, Blocks has to play nicely with other plugins and themes, and WordPress itself. To minimize conflict potential all classes should be prefixed with `.wc-block-`.
```

Comment:
Question from a non-native speaker "To minimize conflict potential" shouldn't be "To minimize potential conflict"? :thinking: 

In addition, I remember @nerrad and @senadir suggested using `.wc-component` for components, I think this doesn't need to be discussed here and can be tackled in the future, but I wanted to point it out so everybody is aware of this.